<launch>

  <!-- defines name of the namespace of the drone -->
  <arg name="uav_name" default="$(optenv UAV_NAME uav)" />

  <arg name="simulation" default="true" />

  <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg uav_name)_ballooncircledestroy_manager" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <!-- Namespace -->
  <group ns="$(arg uav_name)">

    <!-- WaypointFlier nodelet -->
    <node pkg="nodelet" type="nodelet" name="balloon_circle_destroy" args="$(arg nodelet) balloon_circle_destroy/BalloonCircleDestroy $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <!-- Parameters loaded from launch file -->
      <param name="simulation" value="$(arg simulation)" />

      <!-- ROS parameters config file -->
      <rosparam file="$(find balloon_circle_destroy)/config/simulation/simulation_$(arg uav_name).yaml" />

      <!-- Subscribers -->
      <remap from="~odom_uav_in" to="control_manager/cmd_odom" />
      <remap from="~tracker_diagnostics_in" to="control_manager/mpc_tracker/diagnostics" />
      <remap from="~odom_gt_in" to="ground_truth" />
      <remap from="~balloon_point_in" to="balloon_planner/balloon_chosen_out" />
      <remap from="~balloon_point_cloud_in" to="object_detect/detected_objects_pcl" />

      <!-- Publishers -->

      <!-- Service servers -->
      <remap from="~circle_around" to="~circle_around" />
      <!-- Service clients -->
      <remap from="~trajectory_srv" to="control_manager/mpc_tracker/set_trajectory" />
      <remap from="~land_out" to="uav_manager/land" />
      <remap from="~drone_stop" to="control_manager/hover" />
      <remap from="~reset_estimation" to="balloon_planner/reset_estimation" />
      <remap from="~start_estimation" to="balloon_planner/start_estimation" />
      <remap from="~stop_estimation" to="balloon_planner/stop_estimation" />
      <remap from="~add_zone" to="balloon_planner/add_exclusion_zone" />

    </node>

  </group>

</launch>
